python/dolfin/function/expression.py

```diff
@@ -112,4 +114,8 @@ def __init__(self,
         shape: tuple
             Value shape.
         """
-        super().__init__(eval_func, shape)
+        self.shape = shape
+        self._eval_func = eval_func
+
+        eval_address = getattr(eval_func, "address", eval_func)
```

Comment:
Is it necessary to use `getattr`?

Reply:
If supplied `eval_func` has `eval_func.address` then it comes from JIT, otherwise take it as address itself. Its just short alternative to try catch.

Reply:
It's not very clear to me. Better longer and clearer than short and hard to follow.

Reply:
Ok, thats true.